File: src/main/java/it/contrader/dto/ClientDTO.java
package it.contrader.dto;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class ClientDTO {

	private long id;
	
	private String name;
	
	private String surname;
	
	private Date birthdate;
	
	private String telnumber;
	
	private ApartmentDTO apartment;
	
	private TutorDTO tutorDTO;
	
//	private List<DiseaseDTO> diseases;	
	
}

File: src/app/dto/clientdto.ts
import { ApartmentDTO } from './apartmentdto';
import { TutorDTO } from './tutordto';

export class ClientDTO {

    id: number;

    name: string;

    surname: string;

    birthdate: Date;

    telnumber: string ;

    apartmentDTO: ApartmentDTO;

    tutorDTO: TutorDTO;
}
